models/repo.go

```diff
@@ -1553,9 +1553,18 @@ func GetUserMirrorRepositories(userID int64) ([]*Repository, error) {
 }
 
 // GetRecentUpdatedRepositories returns the list of repositories that are recently updated.
-func GetRecentUpdatedRepositories(page, pageSize int) (repos []*Repository, err error) {
-	return repos, x.Limit(pageSize, (page-1)*pageSize).
-		Where("is_private=?", false).Limit(pageSize).Desc("updated_unix").Find(&repos)
+func GetRecentUpdatedRepositories(page, pageSize int, userID int64) (repos []*Repository, err error) {
+	repos, _, err = SearchRepositoryByName(&SearchRepoOptions{
+		Keyword:  "%",
```

Comment:
We can do something like `buildAccessibleRepositoriesQuery` similar to [`buildIssuesQuery`](https://github.com/gogits/gogs/blob/master/models/issue.go#L884), to reuse the same part of both `GetRecentUpdatedRepositories` and `SearchRepositoryByName`.